ledger/sandbox/src/main/scala/com/digitalasset/platform/store/dao/events/TransactionsWriter.scala

```diff
@@ -23,6 +23,33 @@ private[dao] final class TransactionsWriter(
   private val contractsTable = ContractsTable(dbType)
   private val contractWitnessesTable = WitnessesTable.ForContracts(dbType)
 
+  private[dao] final class PreparedInsert(
+      eventBatches: EventsTable.PreparedBatches,
+      flatTransactionWitnessesBatch: Option[BatchSql],
+      complementWitnessesBatch: Option[BatchSql],
+      contractBatches: ContractsTable.PreparedBatches,
+      insertWitnessesBatch: Option[BatchSql],
+  ) {
+    def write()(implicit connection: Connection): Unit = {
+      eventBatches.foreach(_.execute())
+      flatTransactionWitnessesBatch.foreach(_.execute())
+      complementWitnessesBatch.foreach(_.execute())
+
+      for ((deleted, deleteContractsBatch) <- contractBatches.deletions) {
+        val deleteWitnessesBatch = contractWitnessesTable.prepareBatchDelete(deleted.toSeq)
```

Comment:
This slipped in, solving.

Reply:
Addressed by https://github.com/digital-asset/daml/pull/5930/commits/980d99a1cb1337a1df1979506f6c4d0367bb5e22